src/library_pthread.js

```diff
@@ -962,8 +962,12 @@ var LibraryPThread = {
 
   $proxyToMainThread__deps: ['$withStackSave', '_emscripten_run_on_main_thread_js'].concat(i53ConversionDeps),
   $proxyToMainThread__docs: '/** @type{function(number, (number|boolean), ...number)} */',
-  $proxyToMainThread: (index, sync, ...callArgs) => {
-    // Additional arguments are passed after those two, which are the actual
+  $proxyToMainThread: (funcIndex, codePtr, sync, ...callArgs) => {
+    // EM_ASM proxying is done by passing a pointer to the address of the EM_ASM
+    // contant as `codePtr`.  JS library proxying is done by passing and index
+    // into proxiedJSCallArgs as `funcIndex`. If `codePtr` is non-zero then
+    // `funcIndex` will be ignored.
```

Comment:
Could these perhaps be two separate functions, one to proxy em_asm and one for other things? I don't feel strongly though.

Either way, "codePtr" suggests to be an actual C function pointer, but these are indexes in the to EM_ASM array in JS, I think? How about `emAsmIndex`?

Reply:
Its is actually a data pointer pointing the address of the EM_ASM code.    The EM_ASM is an map that is indexed by data pointer.   I guess it could be emAsmAddr?

Reply:
Oh, I see... well, sgtm as is then.

Reply:
But yes, `emAsmAddr` is nicer.